examples/medplum-provider/src/pages/integrations/DoseSpotFavoritesPage.tsx

```diff
@@ -0,0 +1,191 @@
+import { MedicationKnowledge } from '@medplum/fhirtypes';
+import { IconPlus } from '@tabler/icons-react';
+import { formatSearchQuery, normalizeErrorString } from '@medplum/core';
+import { AsyncAutocomplete, useMedplum } from '@medplum/react';
+import { FavoriteMedicationsTable } from './FavoriteMedicationsTable';
+import { showErrorNotification } from '../../utils/notifications';
+import { v4 as uuidv4 } from 'uuid';
+
+/**
+ * This is a demo component for how you could display your favorite Medications
+ * from DoseSpot.
+ *
+ * @returns A React component that displays the favorite medications and allows you to add new favorites.
+ */
+export function DoseSpotFavoritesPage(): React.JSX.Element {
+  const [modalOpened, setModalOpened] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [loadingFavorites, setLoadingFavorites] = useState(true);
+  const [clinicFavoriteMedications, setClinicFavoriteMedications] = useState<MedicationKnowledge[] | undefined>();
+
+  const { state, addFavoriteMedication, searchMedications, setDirections, setSelectedMedication, getMedicationName } =
+    useDoseSpotClinicFormulary();
+  const medplum = useMedplum();
+
+  // Load favorite MedicationKnowledge resources that have a DoseSpot favorite id system
+  useEffect(() => {
+    const loadFavorites = async (): Promise<void> => {
+      try {
+        setLoadingFavorites(true);
+        const searchRequest = {
+          resourceType: 'MedicationKnowledge' as const,
+          filters: [
+            {
+              code: 'code',
+              operator: 'eq' as const,
+              value: `${DOSESPOT_CLINIC_FAVORITE_ID_SYSTEM}|`,
+            },
+          ],
+        };
+        const queryString = formatSearchQuery(searchRequest);
+        const result = await medplum.search('MedicationKnowledge', queryString);
+        setClinicFavoriteMedications(result.entry?.map((e) => e.resource as MedicationKnowledge) || []);
+      } catch (err) {
+        showErrorNotification(err);
+      } finally {
+        setLoadingFavorites(false);
+      }
+    };
+    loadFavorites().catch(showErrorNotification);
+  }, [medplum]);
+
+  const handleAddFavoriteMedication = async (): Promise<void> => {
+    try {
+      setLoading(true);
+      const created = await addFavoriteMedication();
+      // Optimistically update local favorites list
+      setClinicFavoriteMedications((prev) => [created, ...(prev || [])]);
+      setModalOpened(false);
+      showNotification({
+        title: 'Medication added to favorites',
+        message: 'The medication has been added to your favorites',
+        color: 'green',
+      });
+    } catch (error) {
+      showErrorNotification({
+        title: 'Error adding medication to favorites',
+        message: normalizeErrorString(error as Error),
+        color: 'red',
+      });
+    } finally {
+      setSelectedMedication(undefined);
+      setDirections(undefined);
```

Comment:
People will forget to do this. I recommend adding a `clear()` method as well - we probably should have that for the health gorilla hook as well